src/main/java/org/embulk/output/elasticsearch/ElasticsearchOutputPluginDelegate.java

```diff
@@ -0,0 +1,243 @@
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.embulk.base.restclient.RestClientOutputPluginDelegate;
+import org.embulk.base.restclient.RestClientOutputTaskBase;
+import org.embulk.base.restclient.jackson.JacksonServiceRequestMapper;
+import org.embulk.base.restclient.jackson.JacksonTaskReportRecordBuffer;
+import org.embulk.base.restclient.jackson.JacksonTopLevelValueLocator;
+import org.embulk.base.restclient.jackson.scope.JacksonAllInObjectScope;
+import org.embulk.base.restclient.record.RecordBuffer;
+import org.embulk.config.Config;
+import org.embulk.config.ConfigDefault;
+import org.embulk.config.ConfigDiff;
+import org.embulk.config.ConfigException;
+import org.embulk.config.Task;
+import org.embulk.config.TaskReport;
+import org.embulk.spi.Exec;
+import org.embulk.spi.Schema;
+import org.embulk.spi.time.TimestampFormatter;
+import org.embulk.util.retryhelper.jetty92.Jetty92ClientCreator;
+import org.embulk.util.retryhelper.jetty92.Jetty92RetryHelper;
+import org.joda.time.DateTimeZone;
+import org.slf4j.Logger;
+
+import java.util.List;
+import java.util.Locale;
+
+public class ElasticsearchOutputPluginDelegate
+        implements RestClientOutputPluginDelegate<ElasticsearchOutputPluginDelegate.PluginTask>
+{
+    private final Logger log;
+    private final ElasticsearchHttpClient client;
+
+    public ElasticsearchOutputPluginDelegate()
+    {
+        this.log = Exec.getLogger(getClass());
+        this.client = new ElasticsearchHttpClient();
+    }
+
+    public interface NodeAddressTask
+            extends Task
+    {
+        @Config("host")
+        String getHost();
+
+        @Config("port")
+        @ConfigDefault("9200")
+        int getPort();
+    }
+
+    public interface PluginTask
+            extends RestClientOutputTaskBase, TimestampFormatter.Task
+    {
+        @Config("mode")
+        @ConfigDefault("\"insert\"")
+        Mode getMode();
+
+        @Config("nodes")
+        List<NodeAddressTask> getNodes();
+
+        @Config("cluster_name")
+        @ConfigDefault("\"elasticsearch\"")
+        String getClusterName();
+
+        @Config("index")
+        String getIndex();
+        void setIndex(String indexName);
+
+        @Config("alias")
+        @ConfigDefault("null")
+        Optional<String> getAlias();
+        void setAlias(Optional<String> aliasName);
+
+        @Config("index_type")
+        String getType();
+
+        @Config("id")
+        @ConfigDefault("null")
+        Optional<String> getId();
+
+        @Config("bulk_actions")
+        @ConfigDefault("1000")
+        int getBulkActions();
+
+        @Config("bulk_size")
+        @ConfigDefault("5242880")
+        long getBulkSize();
+
+        @Config("concurrent_requests")
+        @ConfigDefault("5")
+        int getConcurrentRequests();
```

Comment:
If you think that you don't need this option, please put @Deprecated in the method and show warning messages if it's used.